### PR TITLE
Newsletter social share 6655

### PR DESCRIPTION
**Pull request by davidtrussler**

### CODE REVIEW

File: app/assets/stylesheets/components/common/_newsletter_signup.scss [hunk 1]
```diff
@@ -6,6 +6,11 @@
   @include body(16, 22);
 }
 
+.t-newsletter-strap {
+  font-size: 1.25rem;
```
Comment (benbarnett) [inline app/assets/stylesheets/components/common/_newsletter_signup.scss]: can you use the `@body` mixin for font sizes? It gives us the px fallback

File: app/assets/stylesheets/components/common/_newsletter_signup.scss [hunk 2]
```diff
@@ -6,6 +6,11 @@
...
+  color: #2e3030;
```
Comment (benbarnett) [inline app/assets/stylesheets/components/common/_newsletter_signup.scss]: Can this be made into a variable?

File: app/assets/stylesheets/components/common/_newsletter_signup.scss [hunk 3]
```diff
@@ -6,6 +6,11 @@
   @include body(16, 22);
 }
 
+.t-newsletter-strap {
```
Comment (benbarnett) [inline app/assets/stylesheets/components/common/_newsletter_signup.scss]: `t-` classes are typically used for testing purposes only. Is that what this is?

File: app/assets/stylesheets/components/common/_newsletter_signup.scss [hunk 4]
```diff
@@ -22,28 +27,18 @@
...
-  .theme-cy & {
-    max-width: 208px;
-
-    @include respond-to($mq-l, $mq-xl) {
-      max-width: 135px;
-    }
-  }
+  padding: $baseline-unit*1.5 $baseline-unit*2; // $baseline-unit*2;
+  margin: 0 auto $baseline-unit*2;
+  max-width: 16.8rem;
```
Comment (benbarnett) [inline app/assets/stylesheets/components/common/_newsletter_signup.scss]: Where does this number come from? We'll also need a non-rem fallback

File: app/assets/stylesheets/components/common/_newsletter_signup.scss [hunk 5]
```diff
@@ -76,13 +71,20 @@
...
-
   &.is-errored {
     @include respond-to($mq-xl) {
       margin-top: -$baseline-unit*2.5;
     }
   }
 }
+
+.t-newsletter-button {
+  min-width: 16.8rem;
```
Comment (benbarnett) [inline app/assets/stylesheets/components/common/_newsletter_signup.scss]: Same here with rem's and `t-` class

**Comment by Tr4pSt3R** [comment]
Hey @davidtrussler has this feature been checkout and used locally by the UX team?

File: app/assets/stylesheets/layout/common/_footer.scss [hunk 6]
```diff
@@ -38,19 +62,164 @@
...
+
+  margin-top: 0;
+  padding-left: 0;
+  text-align: center;
+
+  .footer-primary__list-item {
+    line-height: 1;
+  }
+
+  @include respond-to($mq-s, 37.5em) {
```
Comment (benbarnett) [inline app/assets/stylesheets/layout/common/_footer.scss]: The `37.5em` should probably be referencing one of the 'max' variables in here: https://github.com/moneyadviceservice/yeast/blob/master/assets/lib/_grid.scss#L11-L19

File: app/assets/stylesheets/layout/common/_footer.scss [hunk 7]
```diff
@@ -38,19 +62,164 @@
...
+    @include counter-inline;
+  }
+
+  @include respond-to($mq-xl) {
+    @include counter-inline;
+  }
+
+  .t-twitter-link,
```
Comment (benbarnett) [inline app/assets/stylesheets/layout/common/_footer.scss]: `t-` classnames here

File: app/assets/stylesheets/layout/common/_footer.scss [hunk 8]
```diff
@@ -38,19 +62,164 @@
...
+  }
+}
+
+.social-sharing__counter__total {
+  font-weight: bold;
+  font-size: 1.5rem;
+  display: block;
+
+  @include respond-to($mq-l) {
+    font-size: 1.875rem;
```
Comment (benbarnett) [inline app/assets/stylesheets/layout/common/_footer.scss]: Same here (and above) with font-sizes

Comment (benbarnett) [inline app/assets/stylesheets/layout/common/_footer.scss]: Should be using `@body`

File: app/assets/stylesheets/layout/common/_footer.scss [hunk 9]
```diff
@@ -38,19 +62,164 @@
...
+  }
+}
+
+.social-sharing__counter__label {
+  display: block;
+  font-size: 0.875rem;
```
Comment (benbarnett) [inline app/assets/stylesheets/layout/common/_footer.scss]: same here

File: app/assets/stylesheets/layout/common/_footer.scss [hunk 10]
```diff
@@ -38,19 +62,164 @@
...
+
+    .footer-primary__blog-link,
+    .footer-primary__blog-link:visited,
+    .footer-primary__blog-link:focus,
+    .footer-primary__blog-link:active {
+      background: #0076ab;
+      border-radius: 5px;
+      border-bottom: none;
+      color: white;
+      font-size: 1.125rem;
```
Comment (benbarnett) [inline app/assets/stylesheets/layout/common/_footer.scss]: And here

File: app/assets/stylesheets/layout/common/_footer.scss [hunk 11]
```diff
@@ -38,19 +62,164 @@
...
+      font-weight: 700;
+      padding: 0.75rem 1.5rem;
```
Comment (benbarnett) [inline app/assets/stylesheets/layout/common/_footer.scss]: For top and bottom padding/margins we should use `$baseline-unit` to keep the vertical rhythm

File: app/assets/stylesheets/layout/common/_footer.scss [hunk 12]
```diff
@@ -38,19 +62,164 @@
...
+      @include respond-to($mq-xl) {
+        position: absolute;
+        top: 50%;
+        margin-top: -21px;
+      }
+    }
+
+    .footer-primary__blog-link:hover {
+      text-decoration: none;
+      background-color: #006999;
```
Comment (benbarnett) [inline app/assets/stylesheets/layout/common/_footer.scss]: This should be a colour variable

File: app/assets/stylesheets/components/common/_newsletter_signup.scss (hunk 3 above)
Comment (davidtrussler) [inline app/assets/stylesheets/components/common/_newsletter_signup.scss]: No, this isn't for testing - the t- class was already in use so I just kept it. I'll change that where it appears.

Comment (benbarnett) [inline app/assets/stylesheets/components/common/_newsletter_signup.scss]: Ah right, in which case I think it was there initially as this was purely for A/B testing. As we're now implementing this as a real component, we can safely remove those prefixes.

File: app/assets/stylesheets/components/common/_newsletter_signup.scss [hunk 13]
```diff
@@ -22,28 +27,18 @@
...
-  .theme-cy & {
-    max-width: 208px;
-
-    @include respond-to($mq-l, $mq-xl) {
-      max-width: 135px;
-    }
-  }
+  padding: $baseline-unit*1.5 $baseline-unit*2;
+  margin: 0 auto $baseline-unit*2;
+  max-width: 268px; // 16.8rem;
```
Comment (benbarnett) [inline app/assets/stylesheets/components/common/_newsletter_signup.scss]: You don't have to use this here, but we have a function called `px` that will generate rems or pixel fallback. So you'd use it like this:

```scss
max-width: px(268);
```

But yeah, up to you whether you think it's worth it here

File: app/assets/stylesheets/layout/common/_footer.scss [hunk 14]
```diff
@@ -38,19 +62,161 @@
...
+  @include body(24, 24);
+
+  @include respond-to($mq-l) {
+    @include body(30, 30);
+  }
+}
+
+.social-sharing__counter__label {
+  display: block;
+  // font-size: 0.875rem;
```
Comment (benbarnett) [inline app/assets/stylesheets/layout/common/_footer.scss]: Can be removed

File: app/assets/stylesheets/layout/common/_footer.scss [hunk 15]
```diff
@@ -38,19 +62,161 @@
...
+    .footer-primary__blog-link,
+    .footer-primary__blog-link:visited,
+    .footer-primary__blog-link:focus,
+    .footer-primary__blog-link:active {
+      background: $color-mas-blog-blue;
+      border-radius: 5px;
+      border-bottom: none;
+      color: white;
+      font-weight: 700;
+      padding: $baseline-unit*2 1.5rem;
```
Comment (benbarnett) [inline app/assets/stylesheets/layout/common/_footer.scss]: Probably a good candidate for the `px` function here actually with the horizontal padding

**Comment by benbarnett** [comment]
Nice one :+1:

File: app/assets/stylesheets/components/common/_icon.scss [hunk 16]
```diff
@@ -68,22 +68,72 @@
   height: 26px;
 }
 
-.icon--youtube {
-  background-position: -4px -70px;
-  width: 30px;
-  height: 30px;
+.icon--twitter,
+.svg-icon--twitter {
+  width: 22.149px;
```
Comment (jonalport) [inline app/assets/stylesheets/components/common/_icon.scss]: How come these very precise px values? A pixel is the smallest thing, right?

File: app/assets/stylesheets/layout/common/_footer.scss [hunk 17]
```diff
@@ -38,19 +62,160 @@
...
+    line-height: 1;
+
+    .footer-primary__blog-link,
+    .footer-primary__blog-link:visited,
+    .footer-primary__blog-link:focus,
+    .footer-primary__blog-link:active {
+      background: $color-mas-blog-blue;
+      border-radius: 5px;
+      border-bottom: none;
+      color: white;
```
Comment (jonalport) [inline app/assets/stylesheets/layout/common/_footer.scss]: Should we use a var for this?

**Comment by jonalport** [comment]
One helluva PR. Will there be any squashing? :+1:

File: app/assets/stylesheets/layout/common/_footer.scss (hunk 17 above)
Comment (davidtrussler) [inline app/assets/stylesheets/layout/common/_footer.scss]: Good point. Yes I will add that.

**Comment by benbarnett** [comment]
Rethumbing after squash :+1:

**Comment by jonalport** [comment]
Let's merge this! :ship:

**Pull request merged**